### PR TITLE
[dagster-pagerduty] Migrate to Pythonic resources

### PR DESCRIPTION
**Pull request by benpankow**
## Summary

Migrates the `PagerDutyService` resource to be a Pythonic resource, existing factory method continues to work.

### CODE REVIEW

**Comment by benpankow** [comment]
Current dependencies on/for this PR:
* master
  * **PR #12975** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12975" target="_blank">Graphite</a> 
    * **PR #12977** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12977" target="_blank">Graphite</a>  👈
    * **PR #12976** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12976" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/12977?utm_source=stack-comment).

File: docs/sphinx/sections/api/apidocs/libraries/dagster-pagerduty.rst [hunk 1]
```diff
@@ -30,5 +30,5 @@ alerts from within your ops.
 
 .. currentmodule:: dagster_pagerduty
 
-.. autoconfigurable:: pagerduty_resource
```
Comment (yuhan) [inline docs/sphinx/sections/api/apidocs/libraries/dagster-pagerduty.rst:33]: should we keep the legacy api maybe in a "Legacy" section?

File: python_modules/libraries/dagster-pagerduty/dagster_pagerduty/resources.py [hunk 2]
```diff
@@ -16,8 +20,12 @@ class PagerDutyService:
...
     """
 
-    def __init__(self, routing_key: str):
-        self.routing_key = routing_key
+    routing_key: str = PyField(
+        ...,
+        description="""The routing key provisions access to your PagerDuty service. You
+                    will need to include the integration key for your new integration, as a
+                    routing_key in the event payload.""",
+    )
```
Comment (yuhan) [inline python_modules/libraries/dagster-pagerduty/dagster_pagerduty/resources.py:28]: should this be something like this:
```suggestion
        description=(
            "The routing key provisions access to your PagerDuty service. You",
            "will need to include the integration key for your new integration, as a"
            "routing_key in the event payload."
        ),
    )
```
i haven't checked out the code in my local but i wonder if the triple quotes with indentation would result in a lot whitespaces showing up in the UI?

File: docs/sphinx/sections/api/apidocs/libraries/dagster-pagerduty.rst (hunk 1 above)
Comment (benpankow) [reply docs/sphinx/sections/api/apidocs/libraries/dagster-pagerduty.rst:33]: good idea 👍

**Review by yuhan — APPROVED** [review]

**Pull request merged**